grammar.js

```diff
@@ -209,149 +189,262 @@ module.exports = grammar({
     // tree-sitter to create a "parameters" node in the AST even when
     // no parameters are declared for a function.
+    // Spaces and newlines between the `()` are consumed ahead of time
+    // by the external scanner.
     parameters: $ => seq(
-      "(",
-      optional(commaSep1($, field("parameter", $.parameter))),
-      ")"
+      $._open_parenthesis,
+      optional(seq(
+        field("parameter", $.parameter), 
+        repeat(seq($.comma, field("parameter", $.parameter)))
+      )),
+      $._close_parenthesis
     ),
 
     parameter: $ => choice(
-      seq(field("name", $.identifier), "=", optional(field("default", $._expression))),
-      field("name", choice($.identifier, $.dots))
+      $._parameter_with_default,
+      $._parameter_without_default
+    ),
+
+    _parameter_with_default: $ => seq(
+      field("name", $.identifier), 
+      "=", 
+      optional(field("default", $._expression))
     ),
 
+    _parameter_without_default: $ => field("name", choice($.identifier, $.dots)),
+
     // Control flow.
-    if: $ => prec.right(seq(
+    if_statement: $ => withPrec(PREC.IF, seq(
       "if",
       repeat($._newline),
-      "(",
+      $._open_parenthesis,
       field("condition", $._expression),
-      ")",
+      $._close_parenthesis,
       repeat($._newline),
       field("consequence", $._expression),
       // No `repeat($._newline)` here. Specially handled in the scanner instead.
-      optional(seq(alias($._else, "else"), field("alternative", $._expression)))
+      optional(seq(
+        $._else, 
+        field("alternative", $._expression)
+      ))
     )),
 
-    for: $ => prec.right(seq(
+    for_statement: $ => withPrec(PREC.FUNCTION_OR_LOOP, seq(
       "for",
       repeat($._newline),
-      "(",
+      $._open_parenthesis,
       field("variable", $.identifier),
       "in",
       field("sequence", $._expression),
-      ")",
+      $._close_parenthesis,
       repeat($._newline),
       optional(field("body", $._expression))
     )),
 
-    while: $ => prec.right(seq(
+    while_statement: $ => withPrec(PREC.FUNCTION_OR_LOOP, seq(
       "while",
       repeat($._newline),
-      "(",
+      $._open_parenthesis,
       field("condition", $._expression),
-      ")",
+      $._close_parenthesis,
       repeat($._newline),
       optional(field("body", $._expression))
     )),
 
-    repeat: $ => prec.right(seq(
+    repeat_statement: $ => withPrec(PREC.FUNCTION_OR_LOOP, seq(
       "repeat",
       repeat($._newline),
       optional(field("body", $._expression))
     )),
 
     // Blocks.
-    "{": $ => prec.right(seq(
-      "{",
+    braces: $ => withPrec(PREC.BLOCK, seq(
+      $._open_brace,
       repeat(field("body", choice($._expression, $._semicolon, $._newline))),
-      optional("}")
+      optional($._close_brace)
     )),
 
-    "(": $ => prec.right(seq(
-      "(",
+    parentheses: $ => withPrec(PREC.BLOCK, seq(
```

Comment:
Maybe `paren_expression`? Similar to above, emphasize that this captures an expression within `()`.